### PR TITLE
Xpetra,MueLu,Zoltan2: Fix build errors

### PR DESCRIPTION
**Pull request by cgcgcg**
@trilinos/xpetra @trilinos/muelu @trilinos/zoltan2 

## Motivation
Fixes some issues discovered in an int-int build.

### CODE REVIEW

File: packages/xpetra/src/CrsMatrix/Xpetra_TpetraCrsMatrix_decl.hpp [hunk 1]
```diff
@@ -1249,7 +1249,7 @@ namespace Xpetra {
 
     /// \brief Access the local Kokkos::CrsMatrix data
     typename local_matrix_type::HostMirror getLocalMatrixHost () const {
-      TEUCHOS_UNREACHABLE_RETURN(local_matrix_type::HostMirror());
+      TEUCHOS_UNREACHABLE_RETURN(typename local_matrix_type::HostMirror());
```
Comment (jhux2) [inline packages/xpetra/src/CrsMatrix/Xpetra_TpetraCrsMatrix_decl.hpp:1252]: Why didn't this get caught by existing builds?

File: packages/muelu/src/Utils/MueLu_UtilitiesBase_def.hpp [hunk 2]
```diff
@@ -346,8 +346,8 @@ namespace MueLu {
         using local_vector_type = typename Vector::dual_view_type::t_dev_um;
         using local_matrix_type = typename Matrix::local_matrix_type;
         using execution_space   = typename local_vector_type::execution_space;
-        using rowmap_type       = typename local_matrix_type::row_map_type;
-        using entries_type      = typename local_matrix_type::index_type;
+        // using rowmap_type       = typename local_matrix_type::row_map_type;
+        // using entries_type      = typename local_matrix_type::index_type;
```
Comment (jhux2) [inline packages/muelu/src/Utils/MueLu_UtilitiesBase_def.hpp:350]: just delete?

File: packages/muelu/src/Utils/MueLu_Utilities_def.hpp [hunk 3]
```diff
@@ -580,7 +580,7 @@ namespace MueLu {
         using XCrsMatrix = Xpetra::CrsMatrix<Scalar, LocalOrdinal, GlobalOrdinal, Node>;
         using XCrsMatrixWrap = Xpetra::CrsMatrixWrap<Scalar, LocalOrdinal, GlobalOrdinal, Node>;
         using BCRS = Tpetra::BlockCrsMatrix<Scalar, LocalOrdinal, GlobalOrdinal, Node>;
-        using CRS  = Tpetra::CrsMatrix<Scalar, LocalOrdinal, GlobalOrdinal, Node>;
+        // using CRS  = Tpetra::CrsMatrix<Scalar, LocalOrdinal, GlobalOrdinal, Node>;
```
Comment (jhux2) [inline packages/muelu/src/Utils/MueLu_Utilities_def.hpp:583]: just delete?

File: packages/xpetra/src/CrsMatrix/Xpetra_TpetraCrsMatrix_decl.hpp (hunk 1 above)
Comment (cgcgcg) [reply packages/xpetra/src/CrsMatrix/Xpetra_TpetraCrsMatrix_decl.hpp:1252]: The only reason I can think of is that I built with a newer GCC than before.

**Pull request merged**